### DISCUSSION TITLE
How do I make `docker compose` work when using colima?

### DISCUSSION THREAD

**Discussion by LondonAppDev** [abiosoft/colima#874] [Q&A]
I have Colima installed on macOS.

Running `docker-compose` works fine, however if I run `docker compose`, it says:

 > docker: 'compose' is not a docker command.

I understand that using Docker Desktop will install compose as a plugin so it can be used by running `docker compose`.

Is there a way to set this up when using Colima?

**Comment by valorcz** [comment]
I was in the same situation recently, so just a guess -- you use Homebrew to install docker-compose.

If so, this might help you:

```bash
mkdir -p ~/.docker/cli-plugins
ln -sfn $HOMEBREW_PREFIX/opt/docker-compose/bin/docker-compose ~/.docker/cli-plugins/docker-compose
```

**Reply by rfay** [reply]
And `brew info docker-compose` will tell you this if you don't remember the link here:
```
==> Caveats
Compose is now a Docker plugin. For Docker to find this plugin, symlink it:
  mkdir -p ~/.docker/cli-plugins
  ln -sfn /opt/homebrew/opt/docker-compose/bin/docker-compose ~/.docker/cli-plugins/docker-compose
```

However, if you use `docker-compose` as opposed to `docker compose` there's no need.

**Reply by thiagodeev** [reply]
@rfay you saved me! I'll never forget to read the Homebrew output ever again haha

**Reply by juicygoose** [reply]
@rfay @valorcz you guys are the best danke very much

**Reply by benyamynbrkyc** [reply]
works perfectly

**Reply by gravelfreeman** [reply]
While we're at it here's the one after you installed `homebrew install docker-buildx`:
```bash
ln -sfn "$(brew --prefix)/opt/docker-buildx/bin/docker-buildx" ~/.docker/cli-plugins/docker-buildx
```

**Comment by flavio-fernandes** [comment]
Thank you!

BTW, brew now says this about using docker-compose from colima:

```
==> docker-compose: stable 2.27.1 (bottled), HEAD
Isolated development environments using Docker
https://docs.docker.com/compose/
Installed
/opt/homebrew/Cellar/docker-compose/2.27.1 (8 files, 52.2MB) *
  Poured from bottle using the formulae.brew.sh API on 2024-05-25 at 16:03:37
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/d/docker-compose.rb
License: Apache-2.0
==> Dependencies
Build: go ✘
==> Options
--HEAD
        Install HEAD version
==> Caveats
Compose is a Docker plugin. For Docker to find the plugin, add "cliPluginsExtraDirs" to ~/.docker/config.json:
  "cliPluginsExtraDirs": [
      "/opt/homebrew/lib/docker/cli-plugins"
  ]
```

**Reply by jamesharr** [reply]
FWIW, the old method still seems to work, I wondered into this thread for probably the exact same reason.

**Comment by letrthang** [comment]
for Docker Compose with Colima, still using command: `docker-compose --version` and executable is still `/opt/homebrew/bin/docker-compose`

Not like Docker-desktop which executable for both docker and docker compose are same: `/opt/homebrew/bin/docker`

So if you are using IntelliJ, you need to manually config the path for executable files:

<img width="953" alt="image" src="https://github.com/user-attachments/assets/3a1872e7-cb79-4905-a2e0-0d06b0f280a9">


https://youtrack.jetbrains.com/issue/IDEA-363381/IntelliJ-2024.3-docker-compose-cannot-start-container-unknown-shorthand-flag-f-in-f
